test/integration/oauth_serviceaccount_client_events_test.go

```diff
@@ -151,10 +152,21 @@ func TestOAuthServiceAccountClientEvent(t *testing.T) {
 
 		runTestOAuthFlow(t, testServer, sa, secret, redirect, testCase.expectBadRequest)
 
-		// Check events
-		evList, err := testServer.clusterAdminKubeClient.Core().Events(projectName).List(metav1.ListOptions{})
+		// Check events with a short poll to stop flakes
+		var evList *kapi.EventList
+		err = wait.Poll(time.Second, 5*time.Second, func() (bool, error) {
+			evList, err = testServer.clusterAdminKubeClient.Core().Events(projectName).List(metav1.ListOptions{})
+			if err != nil {
+				return false, nil
```

Comment:
Pretty sure this should be `return false, err`.  The `List` call should never error if I am reading this correctly.